src/mds/Server.cc

```diff
@@ -5894,6 +5903,8 @@ void Server::handle_client_setxattr(MDRequestRef& mdr)
   pi.inode->ctime = mdr->get_op_stamp();
   if (mdr->get_op_stamp() > pi.inode->rstat.rctime)
     pi.inode->rstat.rctime = mdr->get_op_stamp();
+  if (name == "encryption.ctx"sv)
```

Comment:
Is there a typo here? What's up with the "sv" on the end?

In any case, we do need a mechanism like this to handle the case where xattrs are not sent in the trace. In that case, we still need to know that a context is present so we can treat the thing as encrypted. If however, xattrs are always sent in the inode trace, then we may be able to omit this. I wasn't able to tell from looking at the MDS code.

Reply:
> Is there a typo here? What's up with the "sv" on the end?
> 

https://en.cppreference.com/w/cpp/string/basic_string_view/operator%22%22sv

The `sv` is short for `string_view`, with this it will covert the char string to string_view.

One of the benifit of the `sv` here is that it will tolerance for embedded null, etc.

> In any case, we do need a mechanism like this to handle the case where xattrs are not sent in the trace. In that case, we still need to know that a context is present so we can treat the thing as encrypted. If however, xattrs are always sent in the inode trace, then we may be able to omit this. I wasn't able to tell from looking at the MDS code.

So, let keep it for now.

Reply:
Got it. Thanks.